### ISSUE TITLE
Images getting duplicated when clicking and/or dragging-and-dropping quickly

### ISSUE THREAD

**Issue by Sanderovich** [ckeditor/ckeditor5#16926]
## 📝 Provide detailed reproduction steps (if any)

1. Open a CKEditor demo page ([https://ckeditor.com/ckeditor-5/demo/feature-rich/](https://ckeditor.com/ckeditor-5/demo/feature-rich/)).
2. Insert an image.
3. Start clicking and/or dragg-and-dropping the image really fast.

https://github.com/user-attachments/assets/c336e9c2-545c-4a7a-b8a6-6fc282d4c563

### ✔️ Expected result

Nothing happens.

### ❌ Actual result

Image gets duplicated.

## 📃 Other details

* Browser: Firefox version 129.x
* OS: Windows

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by FilipTokarski** [comment]
I was able to reproduce it on the docs demo and on `all-features` manual test as well:
```js
Uncaught TypeError: this._native is null
    get isCanceled datatransfer.ts:113
    _setupDragging dragdrop.ts:314
    fire emittermixin.ts:241
    fireListenerFor bubblingemittermixin.ts:207
    fire bubblingemittermixin.ts:118
    fire domeventobserver.ts:100
    onDomEvent clipboardobserver.ts:98
    observe domeventobserver.ts:76
    fire emittermixin.ts:241
    domListener emittermixin.ts:340
    attach emittermixin.ts:270
    _addEventListener emittermixin.ts:314
    addEventListener emittermixin.ts:894
    listenTo emittermixin.ts:162
    listenTo emittermixin.ts:115
    listenTo emittermixin.ts:112
    observe domeventobserver.ts:74
    observe domeventobserver.ts:73
    attachDomRoot view.ts:317
    init classiceditorui.ts:107
    create classiceditor.ts:194
    promise callback*__webpack_modules__</create/< classiceditor.ts:194
    create classiceditor.ts:191
```

**Comment by Witoso** [comment]
@FilipTokarski Firefox or everywhere?

**Comment by FilipTokarski** [comment]
Just Firefox.

**Comment by Sanderovich** [comment]
@Witoso It looks like the issue is a little bit bigger then I initialy thought. I seem to have a similar issue with Chrome, but you have to be more persistend/ get lucky. In addition, I seem to be able to reproduce the issue quicker when I start drag and dropping images inside each other.

[screencast.webm](https://github.com/user-attachments/assets/84cc8132-3344-4485-bec5-82947706b801)

**Comment by Mati365** [comment]
@kubaklatt Can you check if the issue still happens on Firefox 131? I'm trying to reproduce it but haven't been able to.

**Comment by kubaklatt** [comment]
@Mati365 Yes, the problem still occurs on Firefox 131 - you have to click quickly while moving the mouse

Expected behavior on Chrome:

https://github.com/user-attachments/assets/6ed3520a-fb57-46e3-adc4-1ed3dbc07b09

Actual behavior on Firefox:

https://github.com/user-attachments/assets/7e99ee49-3346-44b0-afa5-9fc458262c4d

**Comment by Mati365** [comment]
I managed to reproduce that on Chrome with CPU 6x Slowdown.
